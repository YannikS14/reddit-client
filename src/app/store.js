import { configureStore } from '@reduxjs/toolkit';
import postsReducer from '../features/posts/postsSlice';
import subredditReducer from '../features/subreddits/subredditsSlice';
import searchFieldReducer from '../features/searchField/searchFieldSlice';

export const store = configureStore({
  reducer: {
    posts: postsReducer,
    subreddits: subredditReducer,
    searchField: searchFieldReducer,
  },
});
