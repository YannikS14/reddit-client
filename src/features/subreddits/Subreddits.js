import React from 'react';
import { useSelector } from 'react-redux';
import { selectSubreddits } from './subredditsSlice';
import Subreddit from '../subreddit/Subreddit';

export default function Subreddits() {
  const { subreddits, activeSubreddit } =
    useSelector(selectSubreddits);

  return (
    <section className="sm:mx-4 sm:w-1/3 sm:ml-0 flex-auto">
      <div className="p-6 bg-white sm:rounded-md shadow-lg">
        <h2 className="text-xl font-bold">Subreddits</h2>
        <div className="mt-4">
          {subreddits.map((subreddit) => {
            return (
              <Subreddit
                subreddit={subreddit}
                key={subreddit.title}
                active={
                  activeSubreddit === subreddit.title ? true : false
                }
              />
            );
          })}
        </div>
      </div>
    </section>
  );
}
