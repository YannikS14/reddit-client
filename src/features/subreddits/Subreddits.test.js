import renderer from 'react-test-renderer';
import { Provider } from 'react-redux';
import { store } from '../../app/store';
import Subreddits from './Subreddits';

describe('Subreddits component', () => {
  it('matches the snapshot', () => {
    const tree = renderer
      .create(
        <Provider store={store}>
          <Subreddits />
        </Provider>,
      )
      .toJSON();
    expect(tree).toMatchSnapshot();
  });
});
