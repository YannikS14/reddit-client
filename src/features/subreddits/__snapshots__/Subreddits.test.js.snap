// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Subreddits component matches the snapshot 1`] = `
<section
  className="lg:mx-4 lg:w-1/3 lg:ml-0 flex-auto"
  role="subreddits"
>
  <div
    className="p-6 bg-white dark:bg-gray-700 sm:rounded-md shadow-lg"
  >
    <h2
      className="text-xl font-bold"
    >
      Subreddits
    </h2>
    <div
      className="mt-4"
    >
      <button
        className="w-full flex items-center p-4 rounded-md transition-colors hover:text-primary bg-gray-100 dark:bg-gray-600"
        onClick={[Function]}
      >
        <img
          alt=""
          className="w-6 h-6 mr-2 border border-gray-500 rounded-full"
          src="https://api.adorable.io/avatars/25/Home"
        />
        <p
          className="font-semibold"
        >
          Popular
        </p>
      </button>
      <button
        className="w-full flex items-center p-4 rounded-md transition-colors hover:text-primary "
        onClick={[Function]}
      >
        <img
          alt=""
          className="w-6 h-6 mr-2 border border-gray-500 rounded-full"
          src="https://b.thumbs.redditmedia.com/EndDxMGB-FTZ2MGtjepQ06cQEkZw_YQAsOUudpb9nSQ.png"
        />
        <p
          className="font-semibold"
        >
          AskReddit
        </p>
      </button>
      <button
        className="w-full flex items-center p-4 rounded-md transition-colors hover:text-primary "
        onClick={[Function]}
      >
        <img
          alt=""
          className="w-6 h-6 mr-2 border border-gray-500 rounded-full"
          src="https://a.thumbs.redditmedia.com/nmh5l-zCsmmc3y2ehfjtWRJjGmCEWEJDTjtW3AGMz60.png"
        />
        <p
          className="font-semibold"
        >
          Wallstreetbets
        </p>
      </button>
      <button
        className="w-full flex items-center p-4 rounded-md transition-colors hover:text-primary "
        onClick={[Function]}
      >
        <img
          alt=""
          className="w-6 h-6 mr-2 border border-gray-500 rounded-full"
          src="https://b.thumbs.redditmedia.com/0PgZl68jAxA6T1BH6uvUQ5Bz1F1GrrJLCL8oi2Gz0Ak.png"
        />
        <p
          className="font-semibold"
        >
          Gaming
        </p>
      </button>
      <button
        className="w-full flex items-center p-4 rounded-md transition-colors hover:text-primary "
        onClick={[Function]}
      >
        <img
          alt=""
          className="w-6 h-6 mr-2 border border-gray-500 rounded-full"
          src="https://b.thumbs.redditmedia.com/Kl3TBjINRBLd9sukJaSPts_0geISdO-jtVniyfCw1GA.png"
        />
        <p
          className="font-semibold"
        >
          CryptoCurrency
        </p>
      </button>
      <button
        className="w-full flex items-center p-4 rounded-md transition-colors hover:text-primary "
        onClick={[Function]}
      >
        <img
          alt=""
          className="w-6 h-6 mr-2 border border-gray-500 rounded-full"
          src="https://a.thumbs.redditmedia.com/4SKK4rzvSSDPLWbx4kt0BvE7B-j1UQBLZJsNCGgMz54.png"
        />
        <p
          className="font-semibold"
        >
          Germany
        </p>
      </button>
      <button
        className="w-full flex items-center p-4 rounded-md transition-colors hover:text-primary "
        onClick={[Function]}
      >
        <img
          alt=""
          className="w-6 h-6 mr-2 border border-gray-500 rounded-full"
          src="https://styles.redditmedia.com/t5_2twpw/styles/communityIcon_8bhyo9k303x71.png"
        />
        <p
          className="font-semibold"
        >
          Blackcats
        </p>
      </button>
      <button
        className="w-full flex items-center p-4 rounded-md transition-colors hover:text-primary "
        onClick={[Function]}
      >
        <img
          alt=""
          className="w-6 h-6 mr-2 border border-gray-500 rounded-full"
          src="https://api.adorable.io/avatars/25/Home"
        />
        <p
          className="font-semibold"
        >
          Cooking
        </p>
      </button>
      <button
        className="w-full flex items-center p-4 rounded-md transition-colors hover:text-primary "
        onClick={[Function]}
      >
        <img
          alt=""
          className="w-6 h-6 mr-2 border border-gray-500 rounded-full"
          src="https://b.thumbs.redditmedia.com/B7IpR8P1mEsQIjdizK5x79s5aGfJUtKk3u2ksGZ9n2Q.png"
        />
        <p
          className="font-semibold"
        >
          Todayilearned
        </p>
      </button>
      <button
        className="w-full flex items-center p-4 rounded-md transition-colors hover:text-primary "
        onClick={[Function]}
      >
        <img
          alt=""
          className="w-6 h-6 mr-2 border border-gray-500 rounded-full"
          src="https://styles.redditmedia.com/t5_2xiqj/styles/communityIcon_zvmcowk0d0671.jpg?format=pjpg&s=1537e0d3fd8639bce301c389ef60e99f44587d0a"
        />
        <p
          className="font-semibold"
        >
          Aoe4
        </p>
      </button>
      <button
        className="w-full flex items-center p-4 rounded-md transition-colors hover:text-primary "
        onClick={[Function]}
      >
        <img
          alt=""
          className="w-6 h-6 mr-2 border border-gray-500 rounded-full"
          src="https://styles.redditmedia.com/t5_2qgzy/styles/communityIcon_rvt3zjh1fc551.png"
        />
        <p
          className="font-semibold"
        >
          Sports
        </p>
      </button>
      <button
        className="w-full flex items-center p-4 rounded-md transition-colors hover:text-primary "
        onClick={[Function]}
      >
        <img
          alt=""
          className="w-6 h-6 mr-2 border border-gray-500 rounded-full"
          src="https://a.thumbs.redditmedia.com/E0Bkwgwe5TkVLflBA7WMe9fMSC7DV2UOeff-UpNJeb0.png"
        />
        <p
          className="font-semibold"
        >
          News
        </p>
      </button>
      <button
        className="w-full flex items-center p-4 rounded-md transition-colors hover:text-primary "
        onClick={[Function]}
      >
        <img
          alt=""
          className="w-6 h-6 mr-2 border border-gray-500 rounded-full"
          src="https://styles.redditmedia.com/t5_2s9h3/styles/communityIcon_ujdrcdijabb61.png"
        />
        <p
          className="font-semibold"
        >
          DunderMifflin
        </p>
      </button>
    </div>
  </div>
</section>
`;
